### DISCUSSION TITLE
Strategies to reduce chunk size

### DISCUSSION THREAD

**Discussion by technoY2K** [nuxt/nuxt#28965] [Questions]
Hello, I'm looking for advice on reducing the chunk size, specifically the one shown in purple below. I've already attempted a few optimizations, including code splitting and configuring the manualChunks function in Vite. Any further suggestions or recommendations would be greatly appreciated.

```
  hooks: {
    'vite:extendConfig'(config, { isClient }) {
      if (isClient) {
        config.build.rollupOptions.output.manualChunks = function (id) {
          // any recommendations please
        };
      }
    },
  }
```

<img width="1659" alt="Screenshot 2024-09-12 at 09 56 46" src="https://github.com/user-attachments/assets/c910e5e3-97c4-4f59-94e0-a7d2e21fa784">

**Comment by joaopedrodcf** [comment]
Hey @technoY2K 

You can try this one maybe it will separate everything a separate chunk, you can then play around and join things that make sense for your application, try to think of cache and how often you change a npm module


```ts
 hooks: {
        "vite:extendConfig"(config, { isClient }) {
            if (isClient) {
                config.build.rollupOptions.output.manualChunks = function (
                    _id: string,
                ) {
                    if (_id.includes("node_modules/")) {
                        const match =
                            /.*node_modules\/((?:@[^\\/]+\/)?[^\\/]+)/.exec(
                                _id,
                            );
                        return match !== null && match.length > 0
                            ? match[1]
                            : "vendor";
                    }
                };
            }
        },
    },
```
